### ISSUE TITLE
Iced_aw Wrap component in libcosmic

### ISSUE THREAD

**Issue by CristianKerr** [pop-os/libcosmic#574]
Hi,
I am migrating my app from Iced (with Iced AW) to libcosmic and I need an alternative for Iced aw Wrap widget.
Its basically a widget that wraps containing components when there is not enough space left. Is there an alternative way of doing this in libcosmic? Or a way to make Iced aw work with libcosmic? (its not working because of iced Element vs cosmic Element and various other things like e.g. I can't use cosmic Length enum with it (I have to use iced_core::length:Length but that is private in libcosmic).

Before somebody asks why do I need it - I have labels that I need to wrap on multiple lines if there is not enough space.

Thank you

**Comment by CristianKerr** [comment]
Now I found flex_row. How do I space the elements inside though? I tried .spacing and .column_spacing but it didn't do it.

**Comment by mmstick** [comment]
Space in what way? There are justify and align methods in addition to setting spacing between rows and columns.

**Comment by CristianKerr** [comment]
I just want there to be exact amount of space between the elements in the flex row. E.g. I want there to be 20px between each element. I thought that that's what spacing is for, but it doesn't work.

**Comment by mmstick** [comment]
Spacing does apply padding between elements. How much space a widget uses though depends on its own length and padding. If it's not working for some reason, it may be a bug.

**Comment by CristianKerr** [comment]
I see, so that is probably a bug. And which one should I use? spacing() or column_spacing()?

**Comment by mmstick** [comment]
One places padding between columns, and the other between rows

**Comment by CristianKerr** [comment]
meaning that the spacing is both column_spacing and row_spacing?

**Comment by mmstick** [comment]
Yes, there is `row_spacing`, `column_spacing`, and `spacing`. See https://pop-os.github.io/libcosmic/cosmic/widget/struct.FlexRow.html
